nf_core/modules/module_test.py

```diff
@@ -0,0 +1,126 @@
+    -------
+    run():
+        Run test steps
+    __check_inputs():
+        Check inputs. Ask for module_name if not provided and check that the directory exists
+    __set_profile():
+        Set software profile
+    __run_pytests(self):
+        Run pytest
+    """
+
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+        self.module_dir = None
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.__check_inputs()
+        self.__set_profile()
+        self.__run_pytests()
+
+    def __check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            if self.no_prompts:
+                raise UserWarning(
+                    f"Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
+                )
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        self.module_dir = os.path.join("modules", *self.module_name.split("/"))
+
+        # First, sanity check that the module directory exists
+        if not os.path.isdir(self.module_dir):
+            raise UserWarning(f"Cannot find directory '{self.module_dir}'. Should be TOOL/SUBTOOL or TOOL")
+
+    def __set_profile(self):
+        """Set $PROFILE env variable.
+        The config expects $PROFILE and Nextflow fails if it's not set.
+        """
+        if os.environ.get("PROFILE") is None:
+            os.environ["PROFILE"] = ""
+            if self.no_prompts:
+                log.info(
+                    "Setting env var '$PROFILE' to an empty string as not set.\n"
+                    "Tests will run with Docker by default. "
+                    "To use Singularity set 'export PROFILE=singularity' in your shell before running this command."
+                )
+            else:
+                question = {
+                    "type": "list",
+                    "name": "profile",
+                    "message": "Choose software profile",
+                    "choices": ["Docker", "Singularity", "Conda"],
+                }
+                answer = questionary.unsafe_prompt([question], style=nf_core.utils.nfcore_question_style)
+                profile = answer["profile"].lower()
+                if profile in ["singularity", "conda"]:
```

Comment:
If PROFILE can be set to docker, and a user specified Docker, it should be set to avoid unexpected behaviour if the default changes at some point in the future. This would also make the if statement unnecessary.

Also, when someone at some point adds a choice, the if statement has to be updated too. So if my proposal to always set the profile after interactively asking for one is not applied, at least the if should be specific for docker instead of the list of profiles that are set. e.g. `if profile != "docker": {set profile}`